src/jit/optimizer.cpp

```diff
@@ -3304,7 +3304,7 @@ bool Compiler::optReachWithoutCall(BasicBlock* topBB, BasicBlock* botBB)
 
     // We can always check topBB and botBB for any gc safe points and early out
 
-    if ((topBB->bbFlags | botBB->bbFlags) & BBF_GC_SAFE_POINT)
+    if ((topBB->bbFlags & BBF_GC_SAFE_POINT) || (!endsWithTailCall && (botBB->bbFlags & BBF_GC_SAFE_POINT)))
```

Comment:
It also should be an error if the sole call in the block is a block ending tail call and we have the flag `BBF_GC_SAFE_POINT` set.

But perhaps we don't yet enforce that invariant, as we may start off considering the call to be a normal call and set the flag, then later on we may opportunistically convert it into a tail call.  We would need to re-walk the statements to see if there is still a different call site to keep the flag set correctly.